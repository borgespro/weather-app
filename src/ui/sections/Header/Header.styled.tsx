import styled from 'styled-components/native';

import { CustomThemeProps } from '@/typings/theme';

export const HeaderContainer = styled.View`
  margin-vertical: ${(props: CustomThemeProps) => props.theme.spacing.xxs};
  justify-content: center;
  align-items: center;
  width: 100%;
`;

export const HeaderContent = styled.View`
  padding-horizontal: ${(props: CustomThemeProps) => props.theme.spacing.xxs};
  padding-vertical: ${(props: CustomThemeProps) => props.theme.spacing.xxs};
  flex-direction: row;
`;

export const AppTitle = styled.Text`
  color: ${(props: CustomThemeProps) => props.theme.colors.title};
  font-size: ${(props: CustomThemeProps) => props.theme.typography.sizes.xxl};
  font-weight: bold;
`;
