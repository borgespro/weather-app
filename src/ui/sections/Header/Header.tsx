import React from 'react';

import SunIcon from '@/ui/elements/SunIcon';

import { AppTitle, HeaderContainer, HeaderContent } from './Header.styled';

export default function Header(): React.ReactElement {
  return (
    <HeaderContainer testID="headerContainer">
      <HeaderContent>
        <SunIcon />
        <AppTitle>WeatherApp</AppTitle>
      </HeaderContent>
    </HeaderContainer>
  );
}
