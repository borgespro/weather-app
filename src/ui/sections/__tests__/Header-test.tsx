import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../Header';

describe('Header Element', () => {
  it('should render without issues', () => {
    const component = shallow(<Header />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
