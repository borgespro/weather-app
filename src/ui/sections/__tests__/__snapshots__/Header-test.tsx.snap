// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Element should render without issues 1`] = `
<Styled(View)
  testID="headerContainer"
>
  <Styled(View)>
    <SunIcon />
    <Styled(Text)>
      WeatherApp
    </Styled(Text)>
  </Styled(View)>
</Styled(View)>
`;
